File: javascript/allergies/allergies.js
const allergens = {
  1: 'eggs',
  2: 'peanuts',
  4: 'shellfish',
  8: 'strawberries',
  16: 'tomatoes',
  32: 'chocolate',
  64: 'pollen',
  128: 'cats'
};

class Allergies {
  constructor(score) {
    this.score = score;
  }

  list() {
    const allergies = [];
    if (this.score === 0) { return allergies; }
    if (allergens.hasOwnProperty(this.score)) { allergies.push(allergens[this.score]); }

    return allergies;
  }
}

export default Allergies;

File: ruby/allergies/allergies.rb
ALLERGENS = {
  128 => 'cats',
  64 => 'pollen',
  32 => 'chocolate',
  16 => 'tomatoes',
  8 => 'strawberries',
  4 => 'shellfish',
  2 => 'peanuts',
  1 => 'eggs'
}.freeze

class Allergies
  attr_reader :list
  def initialize(score)
    @list = ALLERGENS.select { |key, allergen| key & score == key }.values
  end

  def allergic_to?(item)
    @list.include?(item)
  end
end

